x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -312,14 +309,27 @@ public synchronized void triggered(Event event) {
             return;
         }
 
+        if (taskState.get() == DataFrameTransformTaskState.FAILED) {
+            logger.debug("Schedule was triggered for transform [{}] but task is failed. Ignoring trigger.", getTransformId());
+            return;
+        }
+
+        // ignore trigger if indexer is running or completely stopped
+        IndexerState indexerState = getIndexer().getState();
+        if (IndexerState.INDEXING.equals(indexerState) ||
+            IndexerState.STOPPING.equals(indexerState) ||
+            IndexerState.STOPPED.equals(indexerState)) {
```

Comment:
What about `ABORTING`?  Would we want to trigger in this state?

If not then the condition can be `if (indexerState != IndexerState.STARTED)`.

Reply:
`AsyncTwoPhaseIndexer::maybeTriggerAsyncJob` essentially has the same checks I'd be tempted this leave this out. Index state could change between this check and calling `AsyncTwoPhaseIndexer::maybeTriggerAsyncJob`